t/068-socket-keepalive.t

```diff
@@ -1551,3 +1552,1239 @@ done
+            end
+
+            check_opts_for_connect({pool_size = 'a'})
+            check_opts_for_connect({pool_size = 0})
+            check_opts_for_connect({backlog = -1})
+            check_opts_for_connect({backlog = 0})
+        }
+    }
+--- request
+GET /t
+--- response_body_like
+.+ 'connect' \(bad "pool_size" option type: string\)
+.+ 'connect' \(bad "pool_size" option value: 0\)
+.+ 'connect' \(bad "backlog" option value: -1\)
+ok
+--- no_error_log
+[error]
+
+
+
+=== TEST 26: now connect can speficy pool options, which overrides setkeepalive
+--- config
+    location /t {
+        set $port $TEST_NGINX_MEMCACHED_PORT;
+
+        content_by_lua_block {
+            local port = ngx.var.port
+            local function go()
+                local sock = ngx.socket.tcp()
+                local ok, err = sock:connect("127.0.0.1", port, {pool_size = 1})
+                if not ok then
+                    ngx.say("failed to connect: ", err)
+                    return
+                end
+
+                ngx.say("connected: ", ok, ", reused: ", sock:getreusedtimes())
+
+                local req = "flush_all\r\n"
+                local bytes, err = sock:send(req)
+                if not bytes then
+                    ngx.say("failed to send request: ", err)
+                    return
+                end
+                ngx.say("request sent: ", bytes)
+
+                local line, err, part = sock:receive()
+                if line then
+                    ngx.say("received: ", line)
+                else
+                    ngx.say("failed to receive a line: ", err, " [", part, "]")
+                end
+
+                local ok, err = sock:setkeepalive(0, 20)
+                if not ok then
+                    ngx.say("failed to set reusable: ", err)
+                end
+            end
+
+            -- reuse ok
+            go()
+            go()
+
+            local sock1 = ngx.socket.connect("127.0.0.1", port)
+            local sock2 = ngx.socket.connect("127.0.0.1", port)
+            local ok, err = sock1:setkeepalive(0, 20)
+            if not ok then
+                ngx.say(err)
+            end
+            local ok, err = sock2:setkeepalive(0, 20)
+            if not ok then
+                ngx.say(err)
+            end
+
+            -- the pool_size is 1 instead of 20
+            sock1 = ngx.socket.connect("127.0.0.1", port)
+            sock2 = ngx.socket.connect("127.0.0.1", port)
+            ngx.say("reused: ", sock1:getreusedtimes())
+            ngx.say("reused: ", sock2:getreusedtimes())
+            sock1:setkeepalive(0, 20)
+            sock2:setkeepalive(0, 20)
+        }
+    }
+--- request
+GET /t
+--- response_body
+connected: 1, reused: 0
+request sent: 11
+received: OK
+connected: 1, reused: 1
+request sent: 11
+received: OK
+reused: 1
+reused: 0
+--- no_error_log eval
+["[error]",
+"lua tcp socket keepalive: free connection pool for ",
+"lua tcp socket connection pool size: 20"]
```

Comment:
We'd better issue a warning log message for ignoring the pool size specified in the old way (i.e., via `:setkeepalive()` method call's argument).